presto-thrift-interface/src/main/java/com/facebook/presto/thrift/interfaces/client/ThriftPrestoClient.java

```diff
@@ -0,0 +1,159 @@
+     *
+     * @return definitions for session properties supported by Thrift Node.
+     */
+    @ThriftMethod(exception = @ThriftException(type = ThriftServiceException.class, id = 1))
+    List<ThriftPropertyMetadata> listSessionProperties();
+
+    /**
+     * Returns supported schema names.
+     *
+     * @return supported schema names.
+     */
+    @ThriftMethod(exception = @ThriftException(type = ThriftServiceException.class, id = 1))
+    List<String> listSchemaNames();
+
+    /**
+     * Returns tables for the given schema name.
+     *
+     * @param schemaNameOrNull schema name or {@literal null} value.
+     * @return a list of table names with corresponding schemas. If schema name is null then returns
+     * a list of tables for all schemas. Returns an empty list of a schema doesn't exist.
+     */
+    @ThriftMethod(exception = @ThriftException(type = ThriftServiceException.class, id = 1))
+    List<ThriftSchemaTableName> listTables(@Nullable String schemaNameOrNull);
+
+    /**
+     * Returns metadata for a given table.
+     *
+     * @param schemaTableName schema and table name
+     * @return metadata for a given table. If a table doesn't exist then the result should return null value inside.
+     */
+    @ThriftMethod(exception = @ThriftException(type = ThriftServiceException.class, id = 1))
+    ThriftNullableTableMetadata getTableMetadata(ThriftSchemaTableName schemaTableName);
+
+    /**
+     * Returns supported table layouts for a given table, desired columns and query constraint.
+     *
+     * @param session session properties
+     * @param schemaTableName schema and table name
+     * @param outputConstraint query constraint
+     * @param desiredColumns a superset of columns used by the query.
+     * {@literal null} value means "all columns" while empty set means "no columns".
+     * @return supported table layouts
+     */
+    @ThriftMethod(exception = @ThriftException(type = ThriftServiceException.class, id = 1))
+    List<ThriftTableLayoutResult> getTableLayouts(
+            ThriftConnectorSession session,
+            ThriftSchemaTableName schemaTableName,
+            ThriftTupleDomain outputConstraint,
+            @Nullable Set<String> desiredColumns);
+
+    /**
+     * Returns a batch of splits.
+     *
+     * @param session session properties
+     * @param layout table layout chosen by the engine
+     * @param maxSplitCount maximum number of splits to return
+     * @param continuationToken continuation token from a previous split batch or {@literal null} if it's the first call
+     * @return a batch of splits
+     */
+    @ThriftMethod(exception = @ThriftException(type = ThriftServiceException.class, id = 1))
+    ListenableFuture<ThriftSplitBatch> getSplits(
+            ThriftConnectorSession session,
+            ThriftTableLayout layout,
+            int maxSplitCount,
+            @Nullable byte[] continuationToken);
+
+    /**
+     * Returns a batch of table data for the given split.
```

Comment:
Well, those are rows in a columnar format, so I called them "table data". Though, I changed it to "rows" if it's more clear.